docs/guides-adv_invokeconcepts.md

```diff
@@ -0,0 +1,63 @@
+---
+id: adv_invokeconcepts
+title: Advanced - Additional doInvoke functions
+---
+
+In this guide, we will discuss features of `doInvoke` that seem a little more complex. We will add an extra transaction to our invoke with `intents` and externalize the signing of our transaction - so we don't have to pass a user's private key to the config.
+
+## Intents - adding extra transactions
+In the previous `doInvoke` guide, we sent 1 GAS alongside our invocation as a fee. As described [here](http://docs.neo.org/en-us/sc/systemfees.html#smart-contract-fees), transactions with a cost under 10 GAS are essentially free. Unfortunately we can't change the gas parameter of our configuration object to 0, since it has to be an integer value greater than 0.  
```

Comment:
So the `gas` field can be 0.

Reply:
Since the callee wont know exact amount of GAS used, is there a suggested way?

Does it automatically calculate it or is it dependent on the what the callee specifies?

Reply:
@melvin0008 @snowypowers you should use a testinvoke to calculate the GAS cost. I'll add this to the guide?

In the previous `doInvoke` guide, we sent 1 GAS alongside our invocation as a fee. As described [here](http://docs.neo.org/en-us/sc/systemfees.html#smart-contract-fees), transactions with a cost under 10 GAS are essentially free. So our `gas` field should stay 0.

There is a way to achieve this. We can add a transaction to ourselves as an intent, with a minimum amount of 0.00000001 GAS. This makes sure we don't have to add 1 GAS to our transaction (without that GAS being needed to consume the transaction), while making sure a transaction is registered and persisted to the blockchain.

Reply:
Perfect. Thanks a lot for your detailed explanation :)

Reply:
I think mentioning that you can test it with a rpc call is enough. No need to repeat the content. At most, reference with a link.